cli/progress_bar.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use crate::colors;
+use deno_core::parking_lot::Mutex;
+use std::sync::Arc;
+use std::time::Duration;
+
+#[derive(Clone, Debug, Default)]
+pub struct ProgressBar(Arc<Mutex<Option<indicatif::ProgressBar>>>);
+
+impl ProgressBar {
+  fn create() -> indicatif::ProgressBar {
+    let pb = indicatif::ProgressBar::new_spinner();
+    pb.enable_steady_tick(Duration::from_millis(120));
+    pb.set_prefix("Download");
+    pb.set_style(
+      indicatif::ProgressStyle::with_template(
+        "{prefix:.green} {spinner:.green} {msg}",
+      )
+      .unwrap()
+      .tick_strings(&["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"]),
+    );
+    pb
+  }
+
+  pub fn update(&self, msg: &str) {
```

Comment:
I'm not sure I understand. Do you suggest that the progress bar should update to currently in-flight downloads if other download has finished? That would be significantly harder to do, since we don't track what is being downloaded in a single place (in other words, we know when the future is kicked off and when it finishes, but there's no single struct that keeps tabs on all of that). Also I'd have to dig deeper into API to see if it supports such feature.

Reply:
Probably just some API like:

```
let _hold = pb.update("some message");
```

Then when `_hold` is destructed it will pop the message off the pending messages and the progress bar will display the most recent running action.